### PR TITLE
Some Cuda Ico Fixes to Make Vertical Indirection E2E Tests Working

### PR DESCRIPTION
**Pull request by mroethlin**
## Technical Description

This PR addresses a few problems in the code generation for vertical index fields uncovered using the e2e repository.

### Testing

Testing in e2e repo

### Dependencies

This PR is independent.

### CODE REVIEW

**Comment by mroethlin** [comment]
launch jenkins

**Comment by mroethlin** [comment]
launch jenkins

File: dawn/src/dawn/CodeGen/CXXNaive-ico/ASTStencilBody.cpp [hunk 1]
```diff
@@ -305,7 +305,8 @@ void ASTStencilBody::visit(const std::shared_ptr<iir::FieldAccessExpr>& expr) {
         if(expr->getOffset().hasVerticalIndirection()) {
           ss_ << ","
               << "m_" << expr->getOffset().getVerticalIndirectionFieldName() << "(deref(LibTag{}, "
-              << sparseArgName_ << "), k)+" << expr->getOffset().verticalShift() << ")";
+              << sparseArgName_ << "), " << sparseIdx << ", k)+"
+              << expr->getOffset().verticalShift() << ")";
```
Comment (Stagno) [inline dawn/src/dawn/CodeGen/CXXNaive-ico/ASTStencilBody.cpp:309]: Before it was working for a dense IndexField, now only for a sparse IndexField. Both cases need to be codegen'd.

**Review by Stagno — APPROVED** [review]

**Comment by mroethlin** [comment]
launch jenkins

**Comment by mroethlin** [comment]
launch jenkins

**Pull request merged**